specs/rollup-node-p2p.md

```diff
@@ -246,10 +246,13 @@ The extended validator emits one of the following validation signals:
 - `REJECT` score penalties, message is dropped
 
 ## Gossip Topics
+There are two topics for distributing blocks to other nodes faster than proxying through L1 would. These are:
 
-### `blocks`
+### `blocksv1`
+Pre-Canyon/Shanghai blocks are broadcast on `/optimism/<chainId>/0/blocks`.
 
-The primary topic of the L2, to distribute blocks to other nodes faster than proxying through L1 would.
+### `blocksv2`
+Post-Canyon/Shanghai blocks are broadcast on `/optimism/<chainId>/0/blocks`.
 
```

Comment:
This is the same as `blocksv1`?

Reply:
The block content is different between v1 & v2 (withdrawals field). P2P uses SSZ and I think we need to separate because otherwise we would start sending blocks that couldn't be parsed (at least that's why there is the V1/V2 split in the engine API).

I'm not sure if it's easy to have a SSZ parser which tells the difference between a ExecutionPayload with withdrawals versus one without withdrawals.

Reply:
Sorry -- that's a typo in the docs (just fixed). We're setting up two topics [here](https://github.com/ethereum-optimism/optimism/pull/7707/files#diff-03658205f9bcc2e4c45b33bcf3ce3e0e1a194cd6b6399303933ec50abe9bc469R500-R513).